process/block/interceptedBlocks/common.go

```diff
@@ -85,10 +88,35 @@ func checkHeaderHandler(hdr data.HeaderHandler) error {
 	if len(hdr.GetPrevRandSeed()) == 0 {
 		return process.ErrNilPrevRandSeed
 	}
+	err := verifyPreviousBlockProof(hdr, enableEpochsHandler)
+	if err != nil {
+		return err
+	}
 
 	return hdr.CheckFieldsForNil()
 }
 
+func verifyPreviousBlockProof(header data.HeaderHandler, enableEpochsHandler common.EnableEpochsHandler) error {
+	previousAggregatedSignature, previousBitmap := header.GetPreviousAggregatedSignatureAndBitmap()
+	hasProof := len(previousAggregatedSignature) > 0 && len(previousBitmap) > 0
+	hasLeaderSignature := len(previousBitmap) > 0 && previousBitmap[0]&1 != 0
+	isFlagEnabled := enableEpochsHandler.IsFlagEnabled(common.ConsensusPropagationChangesFlag)
```

Comment:
should we check against the header's epoch? What happens if we just activated this feature on the metachain? The metachain will stop notarizing headers that require a couple of rounds to get into the new epoch.

Reply:
added